benchmarks/planetarium_benchmarks.cpp

```diff
@@ -0,0 +1,245 @@
+    Planetarium::Parameters parameters(
+        /*sphere_radius_multiplier=*/1,
+        /*angular_resolution=*/0.4 * ArcMinute,
+        /*field_of_view=*/90 * Degree);
+    return Planetarium(parameters,
+                       perspective,
+                       ephemeris_.get(),
+                       earth_centred_inertial_.get());
+  }
+
+ private:
+  Ephemeris<Barycentric>::FixedStepParameters EphemerisParameters() {
+    return Ephemeris<Barycentric>::FixedStepParameters(
+        SymmetricLinearMultistepIntegrator<QuinlanTremaine1990Order12,
+                                           Position<Barycentric>>(),
+        /*step=*/10 * Minute);
+  }
+
+  Ephemeris<Barycentric>::FixedStepParameters HistoryParameters() {
+    return Ephemeris<Barycentric>::FixedStepParameters(
+        SymmetricLinearMultistepIntegrator<Quinlan1999Order8A,
+                                           Position<Barycentric>>(),
+        /*step=*/10 * Second);
+  }
+
+  not_null<std::unique_ptr<SolarSystem<Barycentric>>> const solar_system_;
+  not_null<std::unique_ptr<Ephemeris<Barycentric>>> const ephemeris_;
+  not_null<MassiveBody const*> const earth_;
+  not_null<std::unique_ptr<NavigationFrame>> const earth_centred_inertial_;
+  DiscreteTrajectory<Barycentric> goes_8_trajectory_;
+};
+
+}  // namespace
+
+void RunBenchmark(benchmark::State& state,
+                  Perspective<Navigation, Camera> const& perspective) {
+  Satellites satellites;
+  Planetarium planetarium = satellites.MakePlanetarium(perspective);
+  RP2Lines<Length, Camera> lines;
+  int total_lines = 0;
+  int iterations = 0;
+  constexpr Instant now = "2000-01-21T04:41:30,5"_TT;
+  while (state.KeepRunning()) {
+    lines = planetarium.PlotMethod2(satellites.GOES8Trajectory().Begin(),
+                                    satellites.GOES8Trajectory().End(),
+                                    now,
+                                    /*reverse=*/false);
+    total_lines += lines.size();
+    ++iterations;
+  }
+  Length min_x = Infinity<Length>();
+  Length min_y = Infinity<Length>();
+  Length max_x = -Infinity<Length>();
+  Length max_y = -Infinity<Length>();
+  int points = 0;
+  for (auto const& line : lines) {
+    points += line.size();
+    for (auto const& point : line) {
+      min_x = std::min(min_x, point.x());
+      min_y = std::min(min_y, point.y());
+      max_x = std::max(max_x, point.x());
+      max_y = std::max(max_y, point.y());
+    }
+  }
+  state.SetLabel(std::to_string(points) + " points in " +
+                 std::to_string(total_lines / iterations) + " lines within [" +
+                 DebugString(min_x) + ", " + DebugString(max_x) + "] × [" +
+                 DebugString(min_y) + ", " + DebugString(max_y) + "]");
+}
+
+void BM_PlanetariumPlotMethod2NearPolarPerspective(benchmark::State& state) {
+  RunBenchmark(state, PolarPerspective(near));
+}
+
+BENCHMARK(BM_PlanetariumPlotMethod2NearPolarPerspective);
```

Comment:
Move all the macros together at the end.

Reply:
Done.